lib/shopping_cart.rb

```diff
@@ -0,0 +1,43 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products
+
+  def initialize (name, capacity)
+    @name = name
+    @capacity = capacity
+    @products = []
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    details = {name: name, capacity: capacity.delete("^0-9")}
```

Comment:
no need to save this into a variable

Reply:
I also think that this deletion your doing is a bit hard coded, I wonder if there's another way you can make sure that no matter what is passed in, you get the number that's at the front?